client/app/views/home_application.coffee

```diff
@@ -59,10 +59,11 @@ module.exports = class ApplicationRow extends BaseView
                     extension = 'png'
                     @icon.removeClass 'svg'
 
-                src = "api/applications/#{app.id}.#{extension}"
+                src = "api/applications/#{@model.id}.#{extension}"
                 @icon.attr 'src', src
                 @icon.show()
                 @icon.removeClass 'stopped'
+                $("##{@model.get 'slug'}-frame").remove()
```

Comment:
Yes, the iframe is created if none can be found for the given application: https://github.com/cozy/cozy-home/blob/master/client/app/views/main.coffee#L234-L271